src/tokeniser_config.py

```diff
@@ -0,0 +1,260 @@
+            Whether to pad the documents to a fixed length. Defaults to True.
+        truncation (bool, optional):
+            Whether to truncate the documents to a fixed length. Defaults to
+            True.
+        max_length (int, optional):
+            Maximum number of tokens in a document. Only relevant if either
+            `padding` or `truncation` is True. Defaults to 512.
+        nfkc_normalisation (bool, optional):
+            Whether to normalise the text using NFKC. Defaults to True.
+        pad_token (str, optional):
+            Token to use for padding. Defaults to '<pad>'.
+        bos_token (str, optional):
+            Token to use for beginning of sentence. Defaults to '<s>'.
+        eos_token (str, optional):
+            Token to use for end of sentence. Defaults to '</s>'.
+        unk_token (str, optional):
+            Token to use for unknown tokens. Defaults to '<unk>'.
+        mask_token (str, optional):
+            Token to use for masking. Defaults to '<mask>'.
+    '''
+    def __init__(self,
+                 tokeniser_type: str,
+                 vocab_size: int,
+                 lower_case: bool,
+                 sentence_piece: bool,
+                 add_prefix_space: bool,
+                 byte_level: bool,
+                 add_sep_and_cls_tokens: bool = True,
+                 padding: bool = True,
+                 truncation: bool = True,
+                 max_length: int = 512,
+                 nfkc_normalisation: bool = True,
+                 pad_token: str = '<pad>',
+                 bos_token: str = '<s>',
+                 eos_token: str = '</s>',
+                 unk_token: str = '<unk>',
+                 mask_token: str = '<mask>'):
+        # Set the configuration
+        self.tokeniser_type = tokeniser_type
+        self.vocab_size = vocab_size
+        self.lower_case = lower_case
+        self.sentence_piece = sentence_piece
+        self.add_prefix_space = add_prefix_space
+        self.byte_level = byte_level
+        self.add_sep_and_cls_tokens = add_sep_and_cls_tokens
+        self.padding = padding
+        self.truncation = truncation
+        self.max_length = max_length
+        self.nfkc_normalisation = nfkc_normalisation
+        self.pad_token = pad_token
+        self.bos_token = bos_token
+        self.eos_token = eos_token
+        self.unk_token = unk_token
+        self.mask_token = mask_token
+
+        # Lower case `tokeniser_type` if possible
+        try:
+            self.tokeniser_type = self.tokeniser_type.lower()
+        except AttributeError:
+            pass
+
+        # Check whether the configuration is valid
+        self.check_config()
+
+    def check_config(self):
+        '''Check that the configuration is valid.
+
+        This both checks that the data types of the inputs are correct, as well
+        as the values.
+
+        Raises:
+            ValueError:
+                If the configuration is invalid.
+        '''
```

Comment:
Have you tried pydantic? HF and spaCy (among others) use it for their base classes.

Reply:
I had never heard about it, but it's very cool! Implemented it now.

Reply:
Well glad to give you some new tools ;)